DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs

```diff
@@ -1461,105 +1603,47 @@ private void GameBroadcastChannel_CTCPReceived(object sender, ChannelCTCPEventAr
                 }
             }
 
+            // Ensure the topic starts with "GAME " as expected
             if (!e.Message.StartsWith("GAME "))
                 return;
 
-            string msg = e.Message.Substring(5); // Cut out GAME part
+            // Extract the game information from the topic string
+            string msg = e.Message.Substring(5); // Cut out "GAME " part
             string[] splitMessage = msg.Split(new char[] { ';' });
 
-            if (splitMessage.Length != 11)
+            // Ensure the message has the expected number of parts
+            if (splitMessage.Length != 14)
             {
-                Logger.Log("Ignoring CTCP game message because of an invalid amount of parameters.");
+                Logger.Log("Ignoring game message because of an invalid number of parameters.");
                 return;
             }
 
+            // We keep this here because this is the fastest way of removing a game
             try
             {
-                string revision = splitMessage[0];
-                if (revision != ProgramConstants.CNCNET_PROTOCOL_REVISION)
-                    return;
-                string gameVersion = splitMessage[1];
-                int maxPlayers = Conversions.IntFromString(splitMessage[2], 0);
-                string gameRoomChannelName = splitMessage[3];
-                string gameRoomDisplayName = splitMessage[4];
-                bool locked = Conversions.BooleanFromString(splitMessage[5].Substring(0, 1), true);
-                bool isCustomPassword = Conversions.BooleanFromString(splitMessage[5].Substring(1, 1), false);
                 bool isClosed = Conversions.BooleanFromString(splitMessage[5].Substring(2, 1), true);
```

Comment:
This won't do, you're currently producing a ton of waste IRC traffic and only read 1% of that waste traffic. This won't do.

Either don't remove the game broadcast messages at all (then the routine of parsing the game info need to be split into a new function that would parse both CTCP and channel topics) or change what is broadcasted and how it is broadcasted. What one could do is to broadcast a message "hey, my options changed, please refetch the game info for channel #123456789" and make the client fetch that channel's topic, though it needs evaluation whether that is better than just broadcasting CTCPs.

@Rampastring wdyt?